### DISCUSSION TITLE
Redirecting webpages

### DISCUSSION THREAD

**Discussion by BetterSquash** [community/community#30878] [New to GitHub]
Hello, my static HTML site is hosted on Gitpages, and I would like to redirect users from webpage A to webpage B.  I could do this via a meta refresh, but I have too many pages and webpage A used to be in its own folder, so that means creating 140 folders with their own index.html file.  I tried a 301 redirect by creating a .htaccess, but it didn't work, so I suppose I need other options.

Any suggestions?

TIA

**Comment by thomas-ayissi** [comment]
Hi TIA,
If your Github Page is coded with PHP or if you can change the language from HTML to PHO, then try this redirection method: https://www.delftstack.com/howto/php/php-redirect/

**Reply by BetterSquash** [reply]
Thanks for the reply.
The site is a simple static html one.

By the way, TIA means Thanks In Advance.

**Comment by ichenglin** [comment]
There's one solution I could think of,
but it sure **shouldn't** be used as a **long-term solution**.

What if you just **delete the whole webpage A folder** in the **production build**,
so it **redirects users** to the **404 webpage**.

Next, we could embed a **Javascript script** in the **404 webpage** that checks on **page load**.
If the user is **redirected from webpage A** to the 404 webpage,
then **redirect the user to webpage B**.

I have created an example of the 404 webpage redirect script I mentioned,
note that it is intended to be **placed in the 404 webpage**.

```js
// example of redirecting https://example.com/A/* to https://example.com/B/*
(() => {
    // ignore all webpages except https://example.com/A/*
    const url_matcher = window.location.pathname.match(/^\/A(\/.*)?$/);
    if (!url_matcher) return;
    // redirect https://example.com/A/* to https://example.com/B/*
    const url_parameters = "/B" + (url_matcher[1] !== undefined ? url_matcher[1] : "");
    window.location.replace(window.location.origin + url_parameters + window.location.search);
})();
```

In addition, I want to mention the test case for the script I provided above.

1. navigate to **[https://github.com/A/some-page?test=1](https://github.com/A/some-page?test=1)**, server responded with **404 not found webpage**.
2. open **console** in **chrome developer tools** (ctrl+shift+i), and **paste in the script** I provided above.
3. the **script redirected** me to **[https://github.com/B/some-page?test=1](https://github.com/B/some-page?test=1)**, it works!

I hope this gives you a better overview of the solution.

**Reply by BetterSquash** [reply]
Thank you so much for taking the time to create the script.
My first worry is that having the link go to a 404 will affect my SEO, so that's a problem.

Secondly, I have  about 140 different links and am unclear how I would adapt the code to fit all those other pages.
And lastly, I want to avoid javascript for any functionality.  The only javascript I have on the site is the google tracking code and if users have javascript turned off, it's not a problem for me.

I was hoping there was something I could do on GitPages that would take care of it before the users even loaded a page.

Again, I really appreciate your taking the time to write this, but I don;'t feel it's the solution for me.

**Reply by ichenglin** [reply]
I see your point,
what if you just **proxy your domain** through **Cloudflare**,
and use its **[page rule redirect feature](https://support.cloudflare.com/hc/en-us/articles/4729826525965-Configuring-URL-forwarding-or-redirects-with-Page-Rules)** to redirect users to webpage B?

**Reply by BetterSquash** [reply]
CloudFlare is a great suggestion, unfortunately on my free plan I only get a handful of redirects and the upgrade to Pro is $20 a month, and I need a free solution.  I think I am going to have to create the folders and redirect via Meta Refresh.  Simple, effective, but time consuming.

**Comment by ichenglin** [comment]
Cant you use wildcard to redirect every webpage A link to webpage B?

**Reply by BetterSquash** [reply]
No, because I used A and B as examples.  Each previous page "website.com/name of-page-here/" would go to a separate "website.com/year/20220827.html" file.  Essentially, I'm moving from a wordpress site to a static HTML site.

**Comment by delosemre** [comment]
Hi, you can redirect with javascript. Sample:

`<script>
  window.location.href = "http://example.com";
</script>`

**Reply by BetterSquash** [reply]
Thanks, but I want to avoid javascript.

**Reply by delosemre** [reply]
Okey, then you can try this.

`<meta http-equiv="refresh" content="0; URL=https://www.example.com/" />`

**Reply by BetterSquash** [reply]
Thanks for trying to help, I do appreciate it, but I don't think you have read my original post and understand the situation.

I have over 140 webpages that need to be redirected. I can do this by creating a folder for each a d putting an index file with the meta refresh in it. But that's a lot of work.

Being able to use a .htaccess file would be easier as that's just one line per webpage, but .htaccess doesn't seem to work on GitPages, so I was asking if there was a similar solution to .htaccess.

If you are wondering why each page has to have a folder, it's because I am converting a Wordpress site to static HTML.

**Comment by delosemre** [comment]
htaccess alternative;

https://nystudio107.com/blog/stop-using-htaccess-files-no-really

**Reply by BetterSquash** [reply]
Thanks, but the .htaccess doesn't even work on GitPages anyway.

**Comment by BetterSquash** [comment]
Hey Everybody, in case anybody is interested, I couldn't find a simple solution, so I just created the 140 or so folders and placed an index.html file inside with the correct Meta Refresh location.  Took a few hours work, but hopefully it was worth it.  Thanks to all who replied and tried to help.
